src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.Split.Tests.cs

```diff
@@ -43,6 +44,8 @@ public static IEnumerable<object[]> Split_TestData()
                 yield return new object[] { engine, "a(?<dot1>.)c(.)e", "123abcde456aBCDe789", RegexOptions.None, 19, 0, new string[] { "123", "d", "b", "456aBCDe789" } };
                 yield return new object[] { engine, "a(?<dot1>.)c(.)e", "123abcde456aBCDe789", RegexOptions.IgnoreCase, 19, 0, new string[] { "123", "d", "b", "456", "D", "B", "789" } };
 
+                yield return new object[] { engine, "a", string.Concat(Enumerable.Repeat("ab", 1000)), RegexOptions.None, 0, 0, new[] { "" }.Concat(Enumerable.Repeat("b", 1000)).ToArray() };
```

Comment:
would it be unacceptably slow to make this close to max string/max array size? that would catch some place where we inadvertently in future, eg., try to allocate twice as much.
but just a nit, feel free to ignore.

Reply:
It would need to be outerloop and would need to deal with frequent OOMs. It's a fine suggestion, but I don't think it's worthwhile.